delivery_unifaun_base/models/delivery.py

```diff
@@ -185,44 +195,48 @@ def _default_contents(self):
                  'line_ids.qty', 'weight_spec')
     def _calculate_weight(self):
         """Calculate the weight of the packages. Includes boxes/pallets etc."""
+
         uom_obj = self.env['uom.uom']
         for package in self:
             # TODO: Does this check work or will it be set to 0 if state != draft?
-            _logger.warn(self.unifaun_id.state)
             if package.unifaun_id.state == 'draft':
-                weight = 0.00
-                for line in package.line_ids:
-                    weight += line.product_qty * line.product_id.weight
-                # Box / Pallet
-                if package.ul_id:
-                    weight += package.ul_id.max_weight
-                # Product distribution packages, e.g. boxes stapled on a pallet.
-                if package.packaging_id:
-                    weight += math.ceil(package.product_qty / package.packaging_id.qty) * package.packaging_id.ul.weight
-                package.weight_calc = weight
-                if package.weight_spec:
-                    package.weight = package.weight_spec
-                else:
-                    package.weight = weight
+                if package.quant_package_id and not package.packaging_id:
+                    package.weight = package.quant_package_id.shipping_weight
+                elif self.packaging_id:
+                    weight = 0.00
+                    for line in package.line_ids:
+                        weight += line.product_qty * line.product_id.weight
+                    # Box / Pallet
+                    if package.quant_package_id:
+                        weight += package.quant_package_id.weight
+                    package.weight_calc = weight
+                    if package.weight_spec:
```

Comment:
Can be rearranged so that we don't do all these calculations if weight_spec is set.